jupyterlab_wipp/src/index.ts

```diff
@@ -17,103 +16,108 @@ export interface INotebookInfo {
-  app.commands.addCommand(registerFileMenuCommandID, {
-    label: 'Register in WIPP',
-    iconClass: 'jp-MaterialIcon jp-LinkIcon',
-    isVisible: () => notebookTracker.currentWidget !== null && notebookTracker.currentWidget === app.shell.currentWidget, // Check if notebook is open to enable menu command
-    execute: () => registerByPath(notebookTracker.currentWidget.context.path)
-  });
+          // Add command to context menu
+          const selectorItem = '.jp-DirListing-item[data-isdir]';
+          app.contextMenu.addItem({
+            command: registerContextMenuCommandID,
+            selector: selectorItem
+          })
 
-  // Add command to the main menu
-  mainMenu.fileMenu.addGroup([
-    {
-      command: registerFileMenuCommandID,
-    }
-  ], 40 /* rank */);
+          //Create command for main menu
+          const registerFileMenuCommandID = 'wipp-register-menu';
+          app.commands.addCommand(registerFileMenuCommandID, {
+            label: 'Register in WIPP',
+            iconClass: 'jp-MaterialIcon jp-LinkIcon',
+            isVisible: () => notebookTracker.currentWidget !== null && notebookTracker.currentWidget === app.shell.currentWidget, // Check if notebook is open to enable menu command
+            execute: () => registerByPath(notebookTracker.currentWidget!.context.path)
+          });
 
-  // Add command to the palette
-  palette.addItem({
-    command: registerFileMenuCommandID,
-    category: 'WIPP',
-    args: {}
-  });
+          // Add command to the main menu
+          mainMenu.fileMenu.addGroup([
+            {
+              command: registerFileMenuCommandID,
+            }
+          ], 40 /* rank */);
 
-  // Create the WIPP sidebar panel
-  const sidebar = new WippSidebar(app, notebookTracker, consoleTracker);
-  sidebar.id = 'wipp-labextension:plugin';
-  sidebar.title.iconClass = 'wipp-WippLogo jp-SideBar-tabIcon';
-  sidebar.title.caption = 'WIPP';
+          // Add command to the palette
+          palette.addItem({
+            command: registerFileMenuCommandID,
+            category: 'WIPP',
+            args: {}
+          });
 
-  // Register sidebar panel with JupyterLab
-  labShell.add(sidebar, 'left', { rank: 200 });
-}
+          // Create the WIPP sidebar panel
+          const sidebar = new WippSidebar(app, notebookTracker, consoleTracker);
+          sidebar.id = 'wipp-labextension:plugin';
+          sidebar.title.iconClass = 'wipp-WippLogo jp-SideBar-tabIcon';
+          sidebar.title.caption = 'WIPP';
 
-/**
- * Initialization data for the jupyterlab_wipp extension.
- */
-const extension: JupyterFrontEndPlugin<void> = {
-  id: 'jupyterlab_wipp',
-  autoStart: true,
-  requires: [ICommandPalette, IMainMenu, INotebookTracker, IFileBrowserFactory, ILabShell, IConsoleTracker],
-  activate: activate
+          // Register sidebar panel with JupyterLab
+          labShell.add(sidebar, 'left', { rank: 200 });
+        }
+      })
+      .catch(reason => {
+        console.error(
```

Comment:
Throw error

Reply:
I compared the effect of throwing new error vs. calling console.error:
```ts
console.error(`The jupyterlab_wipp server extension appears to be missing.\n${reason}`);
throw new Error(`The jupyterlab_wipp server extension appears to be missing.\n${reason}`);
```
They look very similar in the console, but the latter causes unhandled exception since there is nothing up the stack to catch it.
![image](https://user-images.githubusercontent.com/8834829/124190643-78591700-da77-11eb-97e6-9c36d39a70bc.png)
There is no difference to the end user unless they open a console.